### PR TITLE
bf: ZENKO-336 zero fill interval stats if none present

### PR DESCRIPTION
**Pull request by philipyoo**
Re-opened for bert-e. Original PR https://github.com/scality/Arsenal/pull/488

Fixes:
- check if `_zip` receives an empty array
- check if there are data intervals is less than total number of intervals. If so, zero fill empty intervals

### CODE REVIEW

File: lib/metrics/StatsModel.js
Comment (jonathan-gramain) [inline lib/metrics/StatsModel.js]: It does not hurt to set the initial value explicitly, you can keep the last `reduce()` argument.

**Comment by rahulreddy** [comment]
@bert-e wait

**Review by tmacro — DISMISSED** [review]

**Review by jonathan-gramain — DISMISSED** [review]

**Comment by rahulreddy** [comment]
@philipyoo Careful with drop commits. BertE will merge automatically if it has two approvals.

**Comment by jonathan-gramain** [comment]
I know, I approved because you asked him to wait ;) but yes we should take care of this.

What about a rule to prevent bert-e to merge based on regexp to detect drop/squash commits?

**Comment by jonathan-gramain** [comment]
Sorry I realized your comment was for Phil, but I felt involved as well :)

**Comment by rahulreddy** [comment]
It all depends on what @tcarmet can support.

**Review by tmacro — APPROVED** [review]

**Review by rahulreddy — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e help

**Comment by rahulreddy** [comment]
@bert-e status

**Comment by rahulreddy** [comment]
@bert-e force_reset

**Comment by tcarmet** [comment]
Hi @rahulreddy I've created a ticket related to your issue, it's going to be adressed on our next sprint (normally). here's the link https://scality.atlassian.net/browse/BERTE-450

I think I described the need and situation pretty well, but feel free to add a comment on something you would like to have!

File: lib/metrics/StatsModel.js
Comment (rachedbenmustapha) [reply lib/metrics/StatsModel.js]: thanks @jonathan-gramain 
@philipyoo not only does it not hurt, but it was required to fix ZENKO-346 (cloudserver crashing on empty stats, on recent node versions iirc) - so you would have effectively reverted the fix for the crash. It's important to check (with git blame for example) why things are done in a certain way before changing.

**Pull request merged**
